evm/src/main/java/org/hyperledger/besu/evm/code/OpcodesV1.java

```diff
@@ -292,36 +298,57 @@ private OpcodesV1() {
     // static utility class
   }
 
-  static long[] validateAndCalculateJumpDests(final Bytes code) {
+  static String validateCode(final EOFLayout eofLayout) {
+    int sectionCount = eofLayout.getCodeSections().length;
+    return Stream.of(eofLayout.getCodeSections())
+        .map(cs -> validateCode(cs.code, sectionCount))
+        .filter(Objects::nonNull)
+        .findAny()
+        .orElse(null);
+  }
+
+  /**
+   * validates the code section
+   *
+   * @param code the code section code
+   * @return null if valid, otherwise a string containing an error reason.
+   */
+  static String validateCode(final Bytes code, final int sectionCount) {
```

Comment:
Could you create a commit with just these changes so I can cherry-pick it for #4760 and work on top of it? this would help to prevent some conflicts no matter which one gets merged first.
Also, given that we are not using the string response, can we return a boolean and log the failed condition?

Reply:
On a second thought, I think we can keep the strings so we can use those during testing

Reply:
I'm expecting EIP-4200 (#4760) will merge first, then EIP-4750 (#4781), then EIP-5450 (#4805 ), in immediate sequence when it hits mainline.  I've already merged those two into 5450, so you may be able to cherry pick out the changes into 4760.

Reply:
Ideally most RJumpX changes go in 4760, most function changes go into 5450, and all remaining validation goes in 5450.  Since I expect stack and code validation to be two separate passes for the first implementation I think all code based validation can go in the feature PRs.  The stack validation does have some interactions with rjump and callf, but no interactions on the actual code execution parts, but it depends on stack height and such, not stuff like are we rjumping out of bounds.